tests/zuse/crypto/keccak.hoon

```diff
@@ -0,0 +1,333 @@
+      0xc533.dcf8.8cd1.a5df.f22b.914d.3875.bd57.
+        fc17.b2e1.f474.ae36.0c38.77d2
+    ::
+      :-  128
+      0xaf3e.0cc6.e645.01f1.0fd3.9722.e852.355f.
+        d6d8.0d32.1906.31e2.f06c.22ad
+    ::
+      :-  255
+      0x5af5.6987.ea9c.f11f.cd0e.ac5e.bc14.b037.
+        365e.9b11.23e3.1cb2.dfc7.929a
+    ==
+  |-
+  ?~  answers  tst
+  =.  tst
+    %^  expect-eq.tst
+        (keccak-224 p.i.answers (~(got by keccak-inputs) p.i.answers))
+      q.i.answers
+    "keccak-224 {(scow %ud p.i.answers)} bytes"
+  $(answers t.answers)
+::
+++  test-keccak-256
+  ^+  tst
+  =/  answers=(list (pair @ud @))
+    :~
+      :-  0
+      0xc5d2.4601.86f7.233c.927e.7db2.dcc7.03c0.
+        e500.b653.ca82.273b.7bfa.d804.5d85.a470
+    ::
+      :-  1
+      0xeead.6dbf.c734.0a56.caed.c044.696a.1688.
+        7054.9a6a.7f6f.5696.1e84.a54b.d997.0b8a
+    ::
+      :-  4
+      0xb149.e766.d761.2eaf.7d55.f74e.1a4f.dd63.
+        709a.8115.b14f.61fc.d22a.a4ab.c8b8.e122
+    ::
+      :-  8
+      0xe620.d8f2.982b.24fe.daaa.3baa.9b46.c3f9.
+        ce20.4ee3.5666.6553.ecb3.5e15.c3ff.9bf9
+    ::
+      :-  64
+      0x5742.71cd.1395.9e8d.deae.5bfb.db02.a3fd.
+        f54f.2bab.fd0c.beb8.9308.2a97.4957.d0c1
+    ::
+      :-  128
+      0xd82e.257d.000d.c9fa.279a.00e2.961e.3286.
+        d2fe.1c02.ef59.833a.b8a6.a710.1bc2.5054
+    ::
+      :-  255
+      0x348f.b774.adc9.70a1.6b11.0566.9442.625e.
+        6ada.a825.7a89.effd.b5a8.02f1.61b8.62ea
+    ==
+  |-
+  ?~  answers  tst
+  =.  tst
+    %^  expect-eq.tst
+        (keccak-256 p.i.answers (~(got by keccak-inputs) p.i.answers))
+      q.i.answers
+    "keccak-256 {(scow %ud p.i.answers)} bytes"
+  $(answers t.answers)
+::
+++  test-keccak-384
+  ^+  tst
+  =/  answers=(list (pair @ud @))
+    :~
+      :-  0
+      0x2c23.146a.63a2.9acf.99e7.3b88.f8c2.4eaa.
+        7dc6.0aa7.7178.0ccc.006a.fbfa.8fe2.479b.
+        2dd2.b213.6233.7441.ac12.b515.9119.57ff
+    ::
+      :-  1
+      0x1b84.e62a.46e5.a201.8617.54af.5dc9.5c4a.
+        1a69.caf4.a796.ae40.5680.161e.2957.2641.
+        f5fa.1e86.41d7.9583.36ee.7b11.c58f.73e9
+    ::
+      :-  4
+      0xf185.0b2a.bb24.f3fd.683c.7015.8278.9d9e.
+        92b6.a45f.9c34.5f9d.ae7f.7997.c8c9.10e8.
+        8003.e592.e592.81cf.92c9.2d6b.51a1.afd1
+    ::
+      :-  8
+      0x638e.6575.8a29.7cb0.9ded.1ac5.b9e8.f779.
+        8020.00ab.791f.67f3.3c60.be36.4437.93ad.
+        cc8a.4a58.e986.8815.7a41.784f.02a4.bcb2
+    ::
+      :-  64
+      0x14aa.679b.0c11.f9c3.63f5.4933.0261.b45e.
+        1e90.ce31.f4a1.b0ce.5cb9.eb81.bd60.79a3.
+        742d.8602.356c.5098.5d0d.3e54.0fdf.dcfb
+    ::
+      :-  128
+      0x3ade.b7ee.ecf9.069f.143a.1015.1fd4.506a.
+        eef3.a0ef.94ca.65d4.448a.cf1e.892b.8ebb.
+        0887.6318.04dd.64e1.53ad.41fa.e012.7a85
+    ::
+      :-  255
+      0x6bff.1c84.05a3.fe59.4e36.0e3b.ccea.1ebc.
+        d509.310d.c79b.9e45.c263.783d.7a5d.d662.
+        c678.9b18.bd56.7dbd.da15.54f5.bee6.a860
+    ==
+  |-
+  ?~  answers  tst
```

Comment:
This code repeats for each keccak length. It should be a helper function, and we might want to collect the inputs and answers into a single map with the keccak length as the key. Not sure about the latter, but we should not have code repeating like this.

Reply:
Very fair. Also not sure about grouping answers for different hashing functions, so I opted to leave them under their own arms for now.

All done!